### PR TITLE
cp2k 8.1

### PR DESCRIPTION
**Pull request by carlocab**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

**Comment by carlocab** [comment]
Seriously?
```
In file included from /private/tmp/cp2k-20210113-67460-9lu3nu/cp2k-8.1/src/grid/cpu/non_orthorombic_corrections.h:13,
                 from /private/tmp/cp2k-20210113-67460-9lu3nu/cp2k-8.1/src/grid/cpu/non_orthorombic_corrections.c:8:
/private/tmp/cp2k-20210113-67460-9lu3nu/cp2k-8.1/src/grid/cpu/tensor_local.h:11:10: fatal error: malloc.h: No such file or directory
   11 | #include <malloc.h>
      |          ^~~~~~~~~~
compilation terminated.
```

File: Formula/cp2k.rb [hunk 1]
```diff
@@ -74,12 +67,10 @@ def install
...
-    cd "makefiles" do
-      %w[sopt ssmp popt psmp].each do |exe|
-        system "make", "ARCH=Darwin-IntelMacintosh-gfortran", "VERSION=#{exe}"
-        bin.install "../exe/Darwin-IntelMacintosh-gfortran/cp2k.#{exe}"
-        bin.install "../exe/Darwin-IntelMacintosh-gfortran/cp2k_shell.#{exe}"
-      end
+    %w[sopt ssmp popt psmp].each do |exe|
+      system "make", "ARCH=Darwin-IntelMacintosh-gfortran", "VERSION=#{exe}"
+      bin.install "../exe/Darwin-IntelMacintosh-gfortran/cp2k.#{exe}"
+      bin.install "../exe/Darwin-IntelMacintosh-gfortran/cp2k_shell.#{exe}"
```
Comment (carlocab) [inline Formula/cp2k.rb:73]: @fxcoudert do these need changing on ARM? It didn't seem to cause any problems before, but it looks like it's causing errors now?

File: Formula/cp2k.rb [hunk 2]
```diff
@@ -74,12 +67,10 @@ def install
               "-lfftw3", "-lfftw3 -lfftw3_threads"
 
     # Now we build
-    cd "makefiles" do
-      %w[sopt ssmp popt psmp].each do |exe|
-        system "make", "ARCH=Darwin-IntelMacintosh-gfortran", "VERSION=#{exe}"
-        bin.install "../exe/Darwin-IntelMacintosh-gfortran/cp2k.#{exe}"
-        bin.install "../exe/Darwin-IntelMacintosh-gfortran/cp2k_shell.#{exe}"
-      end
+    %w[sopt ssmp popt psmp].each do |exe|
```
Comment (fxcoudert) [inline Formula/cp2k.rb:70]:
```suggestion
    %w[ssmp psmp].each do |exe|
```

File: Formula/cp2k.rb (hunk 1 above)
Comment (fxcoudert) [reply Formula/cp2k.rb:73]: No it's called IntelMacintosh for historical reasons but it's actually arch-agnostic. I believe the errors are generic, and we'll see them on Intel once libint compilation is fixed.

File: Formula/cp2k.rb [hunk 3]
```diff
@@ -23,20 +22,14 @@ class Cp2k < Formula
...
-    url "https://downloads.sourceforge.net/project/libint/v1-releases/libint-1.1.5.tar.gz"
-    sha256 "31d7dd553c7b1a773863fcddc15ba9358bdcc58f5962c9fcee1cd24f309c4198"
-  end
-
-  # Upstream fix for GCC 10, remove in next version
-  # https://github.com/cp2k/dbcsr/commit/fe71e6fe
-  patch do
-    url "https://github.com/Homebrew/formula-patches/raw/0c086813/cp2k/gcc10.diff"
-    sha256 "dfaa319c999d49faae86cafe58ddb3b696f72a89f7cc85acd47b3288c6b9ac89"
+    url "https://github.com/cp2k/libint-cp2k/releases/download/v2.6.0/libint-v2.6.0-cp2k-lmax-4.tgz"
```
Comment (fxcoudert) [inline Formula/cp2k.rb:25]: Let's try to use `https://github.com/cp2k/libint-cp2k/releases/download/v2.7.0-beta.5/libint-v2.7.0-beta.5-cp2k-lmax-4.tgz` instead. I know it says beta, but then that's what they recommend to use along with the latest cp2k version…

**Comment by fxcoudert** [comment]
Patch here: https://github.com/cp2k/cp2k/commit/4139ab6cd2d60c112964a6c75c5d52576ac49e6b

**Comment by fxcoudert** [comment]
I've opened an upstream issue: https://github.com/cp2k/cp2k/issues/1312

**Comment by fxcoudert** [comment]
I'm closing this as "does not compile with latest version of every dependency".

**Comment by dev-zero** [comment]
For the record: "latest version of every dependency" means a a "beta" version of an optional dependency where the last released (non-beta and tested) version of that dependency does not build with the gcc/gfortran shipped with Homebrew due to a defect [in the gcc/gfortran version shipped by Homebrew](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=95889).

**Comment by fxcoudert** [comment]
Sorry if that was unclear: "latest version of every dependency" is the stable, released libint, which is not working. (We tried the beta libint to see what happened, but our goal is to only ship stable, released versions.)

I understand you don't want to work around a compiler bug. But we can't ship an unreleased version of GCC, either. Hence why I closed.

**Comment by dev-zero** [comment]
The latest stable libint is libint 2.6.0, all others are pre-releases (see https://github.com/evaleev/libint/releases) and CP2K builds against that libint 2.6.0 (which is again an optional dependency). libint-2.6.0 again builds with a standards compliant compiler (shipped with every Linux distro, as far as standards go with C-preprocessor in Fortran).

The [patch for gcc](https://github.com/cp2k/cp2k/blob/master/tools/toolchain/scripts/files/gcc-10.2.0-cpp-__has_include.patch) works against a released gcc 10.2, is accepted by upstream and will be part of the next release.

So, instead of fixing a broken compiler you want us to adapt our software to work with a beta version of a dependency?

**Comment by fxcoudert** [comment]
> instead of fixing a broken compiler you want us to adapt our software

I've opened a bug report, if you think this is not worth, you can (and did) close it, it's fine with me. We'll give the cp2k 9 release and hopefully that works better (and gcc 10.3 will be released by then).

**Pull request closed without merging**